packages/tpetra/core/src/Tpetra_Assembly_Helpers.hpp

```diff
@@ -0,0 +1,69 @@
+// @HEADER
+// ***********************************************************************
+//
+//          Tpetra: Templated Linear Algebra Services Package
+//                 Copyright (2008) Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+// @HEADER
+
+#ifndef TPETRA_ASSEMBLY_HELPERS_DECL_HPP
+#define TPETRA_ASSEMBLY_HELPERS_DECL_HPP
+
+namespace Tpetra {
+
+template<typename T>
+void beginFill(T t) { t.beginFill();}
+
+template<typename T>
+void endFill(T t) { t.endFill();}
+
+//! Helper function to begin fill on multiple FE{MultiVector|CrsMatrix} objects at once
+template<typename T, typename... Args>
+void beginFill(T t, Args... args) {
```

Comment:
Pass `T` by reference here.  Also, suppose that we wanted two objects -- say CrsMatrix and Vector -- to share communication during global assembly.  I think that could work here, by overloading `beginFill` and `endFill` as follows:
```c++
template<class SC, class LO, class GO, class NT, class... Args>
void
beginFill (CrsMatrix<SC, LO, GO, NT>& A, MultiVector<SC, LO, GO, NT>& B, Args... args)
{
  special_combined_begin_fill (A, B);
  beginFill (args...);
}
```
Would that be how you might do this?

Reply:
Fixing the reference issue.

As for "combo comm", that's kind of what I'm thinking.  We'd need some sorcery for that, but it would be all template specialization and would not be visible to use code.